matlab/CallHeightGainTerminalCorrectionModel_Validation.c

```diff
@@ -0,0 +1,87 @@
+#include "mex.h"
+
+/*=========================================================================
+ |
+ |  Description:  Validate number and type of inputs for height gain
+ |                terminal correction model
+ |
+ |        Input:  nrhs      - Number of input arguments
+ |                prhs      - Array of pointers to input arguments
+ |                fn_name   - Function name
+ |
+ |       Output:  [void]
```

Comment:
Since this returns nothing and there are no outputs, it would be good to expand the description to say what affects. For example, what happens if validation fails? How would the error be propagated back?